app/models/exam.rb

```diff
@@ -7,9 +7,11 @@ class Exam < ActiveRecord::Base
   belongs_to :guide
   belongs_to :organization
 
-  has_many :authorizations, class_name: 'ExamAuthorization'
+  has_many :authorizations, class_name: 'ExamAuthorization', dependent: :delete_all
   has_many :users, through: :authorizations
 
+  after_destroy { |record| Usage.where(parent_item_id: record.id).destroy_all }
```

Comment:
Aren't we already doing this somewhere?

Perhaps we could implement a method like `Usage.destroy_usages_of(record)`